R/lassnags.r

```diff
@@ -0,0 +1,246 @@
+#' \item{\code{neigh_radii}}{A vector of three radii used in quantifying local-area centered neighborhoods. See 
+#' reference page 171 and Figure 4. Defaults are 1.5, 1, and 2 for the sphere, small cylinder and large cylinder neighborhoods,
+#' respectively}
+#' \item{\code{low_int_thrsh}}{The lower intensity threshold filtering value. See reference page 171. Default is 50}
+#' \item{\code{uppr_int_thrsh}}{The upper intensity threshold filtering value. See reference page 171. Default is 170}
+#' \item{\code{pt_den_req}}{Point density requirement based on plot-level point density defined classes. See reference
+#' page 172. Default is 3}
+#' \item{\code{bbpr_thresholds}}{A 3x4 matrix providing the four average BBPR values for each of the three neighborhood
+#' (sphere, small cylinder and large cylinder neighborhoods) to be used in for conditional assessments and classificaiton
+#' into the following four snag classes: 1) general snag, 2) small snag, 3) vlive crown edge snag, and 4) high canopy 
+#' cover snag. See reference page 172 and Table 2. This matirx must be provided by the user.}
+#' The current implementation is known to use a large amount of memory for storing the N x k integer matrix returning the
+#' near neighbour indices for each point in the point cloud. Improvements are possible in future package versions.}
+#' 
+#' @examples
+#' LASfile <- system.file("extdata", "MixedConifer.laz", package="lidR")
+#' las = readLAS(LASfile, select = "xyzi")
+#' lasnormalize(las, grid_terrain(las, method = "knnidw", k = 10L))
+#' las %<>% lasfilter(Z>1.37, ReturnNumber==1)
+#' 
+#' # Supply a matrix of snag BranchBolePtRatio conditional assessment thresholds (see Wing et al 2015, Table 2, pg. 172)
+#' BBPRthrsh_mat <- matrix(c(0.80, 0.80, 0.70,
+#'                           0.85, 0.85, 0.60,
+#'                           0.80, 0.80, 0.60,
+#'                           0.90, 0.90, 0.55),
+#'                           nrow =3, ncol = 4)
+#' 
+#' # Run snag classificaiton and assign classes to each point    
+#' lassnags(las, algorithm = "wing2015", neigh_radii = c(1.5,1,2), low_int_thrsh = 50, uppr_int_thrsh = 170, pt_den_req = 3, bbpr_thresholds = BBPRthrsh_mat)
+#'     
+#' # Plot it all, tree and snag points...
+#' plot(las, color="SnagCls", colorPalette = rainbow(5))
+#'     
+#' # Filter and plot snag points only 
+#' las %<>% lasfilter(SnagCls>0)
+#' plot(las, color="SnagCls", colorPalette = rainbow(5)[-1])
+#' 
+#' Preform tree segmentation using the watershed method
+#' chm = grid_canopy(lasnorm, res = 0.5, subcircle = 0.2, na.fill = "knnidw", k = 4)
+#' chm = as.raster(chm)
+#' lastrees(lasnorm, "watershed", chm, th = 4)
+#'  
+#' @references
+#' Wing, Brian M.; Ritchie, Martin W.; Boston, Kevin; Cohen, Warren B.; Olsen, Michael J. 2015. 
+#' Individual snag detection using neighborhood attribute filtered airborne lidar data. Remote 
+#' Sensing of Environment. 163: 165-179 https://doi.org/10.1016/j.rse.2015.03.013\cr\cr
+#' 
```

Comment:
Add @author with your name and mine. For all the other function it was implicitly me. But here it is not.